UnitystationLauncher/ViewModels/SecurityPanelViewModel.cs

```diff
@@ -0,0 +1,538 @@
+            }
+
+            File.Copy(sourceFile.FullName,
+                targetFile.FullName, true);
+            Console.WriteLine($"Common file: {file.FullName}");
+        }
+
+        Console.WriteLine("Common files listed.");
+    }
+
+
+    class FileInfoComparer : IEqualityComparer<FileInfo>
+    {
+        public bool Equals(FileInfo? x, FileInfo? y)
+        {
+            if (x == null || y == null) return false;
+            return x.Name.Equals(y.Name, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public int GetHashCode(FileInfo obj)
+        {
+            return obj.Name.GetHashCode();
+        }
+    }
+
+    public void OnScan()
+    {
+        Action<string> info = new Action<string>((string log) => { });
+        Action<string> errors = new Action<string>((string log) => { });
+
+        // Create a DirectoryInfo object for the directory
+        DirectoryInfo directory = new DirectoryInfo(System.AppDomain.CurrentDomain.BaseDirectory);
+
+        DirectoryInfo stagingDirectory = directory.CreateSubdirectory("staging"); //TODO temp
+        DirectoryInfo processingDirectory = directory.CreateSubdirectory("processing");
+        DirectoryInfo? dataPath = null;
+
+        try
+        {
+            DeleteContentsOfDirectory(processingDirectory);
+            info.Invoke("Copying files");
+            CopyFilesRecursively(stagingDirectory.ToString(), processingDirectory.ToString());
+
+            info.Invoke("Cleaning out Dlls and Executables");
+            DeleteFilesWithExtension(processingDirectory.ToString(), "exe");
+            DeleteFilesWithExtension(processingDirectory.ToString(), "dll");
+
+            // Get all files in the directory
+            var directories = processingDirectory.GetDirectories();
+
+
+            // Loop through each file
+            foreach (var directorie in directories)
+            {
+                if (directorie.Name.Contains("_Data"))
+                {
+                    if (dataPath != null)
+                    {
+                        errors.Invoke("oh God 2 Datapaths Exiting!!!");
+                        return;
+                    }
+
+                    dataPath = directorie;
+                }
+            }
+
+            if (dataPath == null)
+            {
+                errors.Invoke("oh God NO Datapath Exiting!!!");
+                return;
+            }
+
+
+            var stagingManaged =
+                stagingDirectory.CreateSubdirectory(Path.Combine(dataPath.Name, "Managed"));
+
+            var dllDirectory = dataPath.CreateSubdirectory("Managed");
+
+            CopyFilesRecursively(stagingManaged.ToString(), dllDirectory.ToString());
+
+
+            DirectoryInfo goodFileCopy = new DirectoryInfo(Path.Combine(System.AppDomain.CurrentDomain.BaseDirectory,
+                "GoodFiles", "VDev", "Managed"));
+
+            info.Invoke("Proceeding to scan folder");
```

Comment:
Same here:

```csharp
Log.Information("Proceeding to scan folder");
```

Reply:
though the reason why its past and is so can be displayed on the UI easily